pkg/provisioner/ironic/testserver/redfish.go

```diff
@@ -0,0 +1,249 @@
+	return m
+}
+
+// GetSystem configures the server with a valid response for /redfish/v1/Systems/<ID>
+func (m *RedfishMock) GetSystem(system string) *RedfishMock {
+
+	var links string
+	switch system {
+	case "NoManageBy":
+		var managedByList, odataCount = buildOdata("ManagedBy", 0, "/redfish/v1/Managers", "BMC-")
+		links = `"Links": {` + managedByList + `` + odataCount + `}`
+	default:
+		var managedByList, odataCount = buildOdata("ManagedBy", 1, "/redfish/v1/Managers", "BMC-")
+		links = `"Links": {` + managedByList + `` + odataCount + `}`
+	}
+	var systemBody = `{
+		"@odata.context": "/redfish/v1/$metadata#ComputerSystem.ComputerSystem",
+		"@odata.id": "/redfish/v1/Systems/System-1",
+		"@odata.type": "#ComputerSystem.v1_5_0.ComputerSystem",
+		"Id": "System-1",
+		` + links + `
+	}`
+	m.ResponseWithCode("/redfish/v1/Systems/System-1", systemBody, http.StatusOK)
+	return m
+}
+
+// GetManager configures the server with a valid response
+// for /redfish/v1/Managers/<ID>
+func (m *RedfishMock) GetManager(manager string) *RedfishMock {
+	var vmedia string
+	switch manager {
+	case "NoVirtualMedia":
+		vmedia = `"VirtualMedia": {}`
+	default:
+		vmedia = `"VirtualMedia": {"@odata.id": "/redfish/v1/Managers/BMC-1/VirtualMedia"}`
+
+	}
+	var managerBody = `{
+		"@odata.context": "/redfish/v1/$metadata#Manager.Manager",
+		"@odata.id": "/redfish/v1/Managers/BMC-1",
+		"@odata.type": "#Manager.v1_3_3.Manager",
+		"Id": "BMC-1",
+		` + vmedia + `
+	  }`
+	m.ResponseWithCode("/redfish/v1/Managers/BMC-1", managerBody, http.StatusOK)
+	return m
+}
+
+// VirtualMedia configures the server with all responses according to the virtual media scenario
+func (m *RedfishMock) VirtualMedia(vmedia string) *RedfishMock {
+	switch vmedia {
+	case "NoCD":
+		m.ListVirtualMedia(1)
+		m.VirtualMedia1()
+	case "NoActions":
+		m.ListVirtualMedia(2)
+		m.VirtualMedia1()
+		m.VirtualMedia2(false)
+	default:
+		m.ListVirtualMedia(2)
+		m.VirtualMedia1()
+		m.VirtualMedia2(true)
+	}
+	return m
+}
+
+// ListVirtualMedia configures the server with a valid response for /redfish/v1/Systems with
+// a specific number of system members
+func (m *RedfishMock) ListVirtualMedia(vmediaMembers int) *RedfishMock {
+	membersList, odata := buildOdata("Members", vmediaMembers, "/redfish/v1/Managers/BMC-1/VirtualMedia", "")
+	m.ResponseWithCode("/redfish/v1/Managers/BMC-1/VirtualMedia", `
+	{
+		"@odata.context": "/redfish/v1/$metadata#VirtualMediaCollection.VirtualMediaCollection",
+		"@odata.id": "/redfish/v1/Managers/BMC-1/VirtualMedia",
+		"@odata.type": "#VirtualMediaCollection.VirtualMediaCollection",
+		"Description": "Virtual Media Services Settings",
+		`+membersList+`
+		`+odata+`,
+		"Name": "Virtual Media Services"
+	}`, http.StatusOK)
+	return m
+}
+
+// VirtualMedia1 configures the server with a valid responses for redfish/v1/Managers/BMC-1/VirtualMedia/1
+// this virtual media represents a USB
+func (m *RedfishMock) VirtualMedia1() *RedfishMock {
```

Comment:
I have some concerns on having methods like `VirtualMedia1()` and `VirtualMedia2()` directly defined on `RedfishMock`: the mock should be as much as possible generic, while the creation of actual answer should be delegated to the specific test, which knows exactly all the details of the scenario it's working on. If needed, builder methods (ie like `makeHost`) could be defined - to make simple setting up a given scenario - but still in the test fixture
code

Reply:
Sure, I will change that

Reply:
@andfasano I've created an struct in the tests to help sending the information to generate a virtual media response based on the information.